src/core/net/udp6.cpp

```diff
@@ -297,12 +342,40 @@ otError Udp::SendTo(SocketHandle &aSocket, Message &aMessage, const MessageInfo
 
 void Udp::AddSocket(SocketHandle &aSocket)
 {
-    IgnoreError(mSockets.Add(aSocket));
+    SuccessOrExit(mSockets.Add(aSocket));
+
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
+    if (mPrevBackboneSockets == nullptr)
+    {
+        mPrevBackboneSockets = &aSocket;
+    }
+#endif
+exit:
+    return;
 }
 
 void Udp::RemoveSocket(SocketHandle &aSocket)
 {
     SuccessOrExit(mSockets.Remove(aSocket));
+
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
+    if (&aSocket == mPrevBackboneSockets)
+    {
+        SocketHandle *backboneSockets = aSocket.GetNext();
+        otError       error;
+
+        if (backboneSockets != nullptr)
+        {
+            IgnoreError(error = mSockets.Find(*backboneSockets, mPrevBackboneSockets));
```

Comment:
Remove `IgnoreError()`?

Reply:
If we remove `IgnoreError()` , we need to add `OT_UNUSED_VARIABLE(error)` (note that `OT_ASSERT` can be turned off). 
I am fine with both, but `IgnoreError()` reduced one line of code.

Reply:
**Just for discussion:** 
If we have a method like 
```c++
static inline void OT_ASSERT_OK(otError aError) 
{
    OT_UNUSED_VARIABLE(aError);
    OT_ASSERT(aError == OT_ERROR_NONE);
}
```
 , we can avoid using `IgnoreError` or `OT_UNUSED_VARIABLE`. It's quite easy (and a little annoying) to miss these statements and get build failures.

Reply:
I don' think that would work on all toolchains (the `OT_UNUSED_VARIABLE` can apply to local parameter variable `aError` and not the caller one). 

The trickiness in general is that `OT_UNUSED_VARIABLE` is tool-chain dependent.

 I would recommend using `OT_UNUSED_VARIABLE` directly to silence the unused-var warning from such `OT_ASEERT()` use and not use `IgnoreError` for this. `IgnoreError` is to ensure returned value is not ignored and it just happens to work in this case to use the variable (silence the warning), but may not work in all situations and toolchains.

Reply:
**Note that the code we are commenting is already removed, so resolving this.**